app/v1/models/BirthDateVerification.scala

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v1.models
+
+import play.api.libs.json.{JsString, Json, Reads, Writes, __}
+
+sealed trait BirthDateVerification{
+  val value: String
+}
+
+object BirthDateVerification {
+  private def valueCheck(input: String): BirthDateVerification = {
+    input match {
+      case Unverified.value => Unverified
+      case Verified.value => Verified
+      case VerificationNotKnown.value => VerificationNotKnown
+      case CoegConfirmed.value => CoegConfirmed
+      case _ => throw new IllegalArgumentException(s"Unable to parse $input into a valid value. Available values are: " +
```

Comment:
Can you remove `input` from this please. Can you add a Logger.warn as well for this please. I don't think we should list the available values either